src/ring_theory/unique_factorization_domain.lean

```diff
@@ -1333,6 +1374,23 @@ begin
     cases mul_eq_zero.mp h; contradiction }
 end
 
+/-- The only divisors of prime powers are prime powers. -/
+theorem eq_pow_find_of_dvd_irreducible_pow {a p : associates α} (hp : irreducible p)
+  [∀ n : ℕ, decidable (a ∣ p ^ n)] (h : ∃ n : ℕ, a ∣ p ^ n) : a = p ^ nat.find h :=
+begin
+  nontriviality α,
+  letI := classical.dec_eq α,
+  letI := classical.dec_eq (associates α),
+  letI := λ p : associates α, classical.dec (irreducible p),
+  convert eq_pow_count_factors_of_dvd_pow hp h,
```

Comment:
It seems like the rest of the proof is proving `nat.find h = p.count a.factors`. Could you please extract that as a further sublemma, and then simply rw with that?